### PR TITLE
FiscalYear + DuesPackage model changes

### PR DESCRIPTION
**Pull request by kberzinch**
Summary of changes:
- Add `FiscalYear` model and Nova resource
- Add relationship to `DuesPackage`s and Nova action to bulk-create `DuesPackage`s for a `FiscalYear`
- Add `restricted_to_students` field to `DuesPackage`
- Only show `DuesPackages` that meet new criteria on the dues flow - note that this is a purely cosmetic change and someone could abuse the API to purchase a different package if they wanted to

Known remaining work:
- [x] Figure out dues package conflict representation #610 #1484 
- [x] Figure out affiliation restrictions #1162 

Out of scope:
- Merch changes

### CODE REVIEW

File: app/Models/DuesPackage.php [hunk 1]
```diff
@@ -97,30 +101,70 @@ class DuesPackage extends Model
...
     /**
      * Scope a query to only include DuesPackages available for purchase.
      */
     public function scopeAvailableForPurchase(Builder $query): Builder
     {
-        return $query->where('available_for_purchase', 1);
+        return $query->where('available_for_purchase', true);
+    }
+
+    public function scopeUserCanPurchase(Builder $query, User $user): Builder
```
Comment (RossIV) [inline app/Models/DuesPackage.php:150]: Does query chaining work properly with this?

File: app/Models/DuesPackage.php [hunk 2]
```diff
@@ -97,30 +101,70 @@ class DuesPackage extends Model
...
+            })
+            ->leftJoin('payments', static function (JoinClause $join): void {
+                $join->on('payments.payable_id', '=', 'dues_transactions.id')
+                        ->where('payments.payable_type', '=', DuesTransaction::getMorphClassStatic())
+                        ->where('payments.deleted_at', '=', null)
+                        ->where('payments.amount', '>', 0);
+            })
+            ->where('available_for_purchase', true)
+            ->where('dues_packages.effective_end', '>=', date('Y-m-d'))
+            ->where('restricted_to_students', 'student' === $user->primary_affiliation)
```
Comment (RossIV) [inline app/Models/DuesPackage.php:166]: are we masking `credit-student` into `student`?

Comment (kberzinch) [reply app/Models/DuesPackage.php:166]: We don't see `credit-student`

File: app/Models/DuesPackage.php (hunk 1 above)
Comment (kberzinch) [reply app/Models/DuesPackage.php:150]: Probably? It's not any worse than the `paid()` scope which seems to play nice with others.

**Review by RossIV — APPROVED** [review]

File: app/Nova/FiscalYear.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * Get the fields displayed by the resource.
+     *
+     * @return array<\Laravel\Nova\Fields\Field>
+     */
+    public function fields(Request $request): array
+    {
+        return [
+            Number::make('Ending Year')
+                ->rules('required', 'max:255')
```
Comment (JoshOldenburg) [inline app/Nova/FiscalYear.php:46]: This should ensure the year is >=2000 or similar so people don't enter two digit numbers.

**Review by JoshOldenburg — CHANGES_REQUESTED** [review]
Looks good. One minor change

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**